types/index.d.ts

```diff
@@ -1984,7 +2008,24 @@ declare module 'mongoose' {
     [key: string]: any;
   };
 
-  type ApplyBasicQueryCasting<T> = T | T[] | any;
+  export interface ObjectIdLike {
+    toHexString(): string;
+  }
+  type Impossible = never;
+
+  type LoosenType<R> = R extends infer T ?
+    [T] extends [string] ? T | RegExp :
+      [T] extends [ObjectIdLike] ? T | ObjectIdLike | { _id: ObjectIdLike | string } | string :
+        [T] extends [globalThis.Date] ? T | number :
```

Comment:
We do also allow certain strings for dates

Reply:
Easily added, I'll do that